intermediate_source/torch_export_tutorial.py

```diff
@@ -0,0 +1,533 @@
+# by tracing the target function, given example inputs.
+#
+# The signature of ``torch.export`` is:
+#
+# .. code:: python
+#
+#     export(
+#         f: Callable,
+#         args: Tuple[Any, ...],
+#         kwargs: Optional[Dict[str, Any]] = None,
+#         *,
+#         constraints: Optional[List[Constraint]] = None
+#     ) -> ExportedProgram
+#
+# ``torch.export`` traces the tensor computation graph from calling ``f(*args, **kwargs)``
+# and wraps it in an ``ExportedProgram``, which can be serialized or executed later with
+# different inputs. Note that while the output ``ExportedGraph`` is callable, it is not a
+# ``torch.nn.Module``. We will detail the ``constraints`` argument later in the tutorial.
+
+import torch
+from torch.export import export
+
+class MyModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.lin = torch.nn.Linear(100, 10)
+
+    def forward(self, x, y):
+        return torch.nn.functional.relu(self.lin(x + y), inplace=True)
+
+mod = MyModule()
+exported_mod = export(mod, (torch.randn(8, 100), torch.randn(8, 100)))
+print(type(exported_mod))
+print(exported_mod(torch.randn(8, 100), torch.randn(8, 100)))
+
+######################################################################
+# Let's review some attributes of ``ExportedProgram`` that are of interest.
+#
+# The ``graph`` attribute is an `FX graph <https://pytorch.org/docs/stable/fx.html#torch.fx.Graph>`__
+# traced from the function we exported, that is, the computation graph of all PyTorch operations.
+# The FX graph has some important properties:
+#
+# - The operations are "ATen-level" operations.
+# - The graph is "functionalized", meaning that no operations are mutations.
+#
+# The ``graph_module`` attribute is the ``GraphModule`` that wraps the ``graph`` attribute
+# so that it can be ran as a ``torch.nn.Module``.
+# We can use ``graph_module``'s ``print_readable``` to print a Python code representation
+# of ``graph``:
+
+print(exported_mod)
+exported_mod.graph_module.print_readable()
+
+######################################################################
+# The printed code shows that FX graph only contains ATen-level ops (such as ``torch.ops.aten``)
+# and that mutations were removed. For example, the mutating op ``torch.nn.functional.relu(..., inplace=True)``
+# is represented in the printed code by ``torch.ops.aten.relu.default``, which does not mutate.
+# Future uses of input to the original mutating ``relu`` op are replaced by the additional new output
+# of the replacement non-mutating ``relu`` op.
+#
+# Other attributes of interest in ``ExportedProgram`` include:
+#
+# - ``graph_signature`` -- the inputs, outputs, parameters, buffers, etc. of the exported graph.
+# - ``range_constraints`` and ``equality_constraints`` -- constraints, covered later
+
+print(exported_mod.graph_signature)
+print(exported_mod.range_constraints)
+print(exported_mod.equality_constraints)
```

Comment:
I don't think we should print these here, because they're empty. It's more useful if we print it with the dynamic_dims tracing.